src/coreclr/ToolBox/superpmi/superpmi-shared/lightweightmap.h

```diff
@@ -385,11 +385,9 @@ class LightWeightMap : public LightWeightMapBuffer
 
         if (numItems > 0)
         {
-            for (unsigned int i = numItems; i > insert; i--)
-            {
-                pKeys[i]  = pKeys[i - 1];
-                pItems[i] = pItems[i - 1];
-            }
+            int countToMove = (numItems - insert);
+            memmove(&pKeys[insert+1], &pKeys[insert], countToMove * sizeof(pKeys[insert]));
```

Comment:
Using `memmove` makes sense, but shouldn't this be:
```
int countToMove = numItems - insert;
if (countToMove > 0)
{
  memmove(&pKeys[numItems], &pKeys[numItems - 1], countToMove * sizeof(pKeys[insert]));
...
```

As it is, it seems like it will only work if inserting just before the very last element, or inserting as the last element (when `countToMov == 0`).

Reply:
I don't think I understand, I think the current implementation is right
![image](https://user-images.githubusercontent.com/22642771/121433430-d723e880-c930-11eb-97a6-d33031290114.png)

Reply:
You're right... I was thinking in some weird backwards world...